lib/FusionInventory/Agent/Tools/Hardware.pm

```diff
@@ -1401,35 +1403,82 @@ sub _getVlans {
         }
     }
 
-    # For other switches, we use another methods
-    # used for Alcatel-Lucent and ExtremNetworks (and perhaps others)
-    my $vlanIdName = $snmp->walk('.1.0.8802.1.1.2.1.5.32962.1.2.3.1.2');
-    my $portLink = $snmp->walk('.1.0.8802.1.1.2.1.3.7.1.3');
-    if($vlanIdName && $portLink){
-        foreach my $suffix (sort keys %{$vlanIdName}) {
-            my ($port, $vlan) = split(/\./, $suffix);
-            if ($portLink->{$port}) {
-                # case generic where $portLink = port number
-                my $portnumber = $portLink->{$port};
-                # case Cisco where $portLink = port name
-                unless ($portLink->{$port} =~ /^[0-9]+$/) {
-                    $portnumber = $port;
+    # For Switch with dot1qVlanStaticEntry and dot1qVlanCurrent Present
+    my $dot1qVlanStaticName = $snmp->walk('.1.3.6.1.2.1.17.7.1.4.3.1.1');
+    my $dot1qVlanStaticRowStatus = $snmp->walk('.1.3.6.1.2.1.17.7.1.4.3.1.5');
+    # each result matches either of the following schemes :
+    # $prefix.$i    = $value with $i as vlan_id, and each bit of $value represent the Egress (or Untagged) is present (1st bit = ifnumber 1, 2nd bit => ifnumber 2, etc...)
+    my $dot1qVlanCurrentEgressPorts = $snmp->walk('.1.3.6.1.2.1.17.7.1.4.2.1.4');
+    my $dot1qVlanCurrentUntaggedPorts = $snmp->walk('.1.3.6.1.2.1.17.7.1.4.2.1.5');
+
+    if($dot1qVlanStaticRowStatus and $dot1qVlanStaticRowStatus and $dot1qVlanCurrentEgressPorts and $dot1qVlanCurrentUntaggedPorts){
+        foreach my $vlan_id (sort keys %{$dot1qVlanStaticRowStatus}) {
+            if($dot1qVlanStaticRowStatus->{$vlan_id} eq 1) {
+                my $name = $dot1qVlanStaticName->{$vlan_id};
+
+                my $suffix = defined($dot1qVlanCurrentEgressPorts->{$vlan_id}) ? $vlan_id : ("0.".$vlan_id);
+                next if !defined($dot1qVlanCurrentEgressPorts->{$suffix});
+
+                # Tagged & Untagged VLAN
+                my $bEgress = '';
+                if(Math::BigInt->from_hex($dot1qVlanCurrentEgressPorts->{$suffix}) ne Math::BigInt->bnan()){
```

Comment:
I would prefer another way of doing in place of introducing a new dependency on Math::BigInt here and on l.1432.
Can you justify this ?
You should also check in `FusionInventory::Agent::Tools` or `FusionInventory::Agent::Tools::SNMP` if some method can still be used. Or eventually add ones.